llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1456,38 +1456,60 @@ InstructionCost VPWidenRecipe::computeCost(ElementCount VF,
 
 void VPWidenEVLRecipe::execute(VPTransformState &State) {
   unsigned Opcode = getOpcode();
-  // TODO: Support other opcodes
-  if (!Instruction::isBinaryOp(Opcode) && !Instruction::isUnaryOp(Opcode))
-    llvm_unreachable("Unsupported opcode in VPWidenEVLRecipe::execute");
+  if (Opcode == Instruction::ICmp || Opcode == Instruction::FCmp) {
```

Comment:
Shall we instead represent it as WPWidentIntrisic?

Reply:
If possible that would be great, given that there are quite a few changes to the existing code here

Reply:
pls see: https://github.com/llvm/llvm-project/pull/114205/files